lib/tests/test_working_copy.rs

```diff
@@ -210,73 +212,120 @@ fn test_checkout_file_transitions(use_git: bool) {
     wc.check_out(repo.op_id().clone(), None, &right_tree)
         .unwrap();
 
-    // Check that the working copy is clean.
+    // Check that the working copy is clean other than the submodules
+    // that are modified to a different type.
     let mut locked_wc = wc.start_mutation();
     let new_tree_id = locked_wc.snapshot(GitIgnoreFile::empty()).unwrap();
     locked_wc.discard();
-    assert_eq!(new_tree_id, right_tree_id);
 
-    for (_left_kind, right_kind, path) in &files {
+    if use_git {
+        let new_tree = repo
+            .store()
+            .get_tree(&RepoPath::root(), &new_tree_id)
+            .unwrap();
+        let diff_snapshot = new_tree.diff_summary(
+            &repo
+                .store()
+                .get_tree(&RepoPath::root(), &right_tree_id)
+                .unwrap(),
+            &matchers::EverythingMatcher,
+        );
+        assert_eq!(
+            diff_snapshot,
+            DiffSummary {
+                added: vec![
+                    "GitSubmodule_Conflict",
+                    "GitSubmodule_Executable",
+                    "GitSubmodule_ExecutableNormalContent",
+                    "GitSubmodule_Normal",
+                    "GitSubmodule_Symlink",
```

Comment:
As per my understanding this should also have `GitSubmodule_Missing`, but I don't know why it doesn't show up in the added here. Manually running the command and testing it, by reverting to a commit that doesn't have submodule object results in the files in the submodule added here